### DISCUSSION TITLE
Anybody having trouble removing cookies?

### DISCUSSION THREAD

**Discussion by E8y2FqZE** [sqlpage/SQLPage#361] [Q&A] [answered]
I've made a "flash message" using the alert component that I want to show exactly once when someone successfully updates a resource. So for example if I set a cookie like this:

```sql
update users set
	name = :name,
	email = :email
	where id = :id;

select 'cookie' as component,
	'admin.alert.title' as name,
	'User updated' as value;

select 'redirect' as component,
	'user.sql?user=' || :id as link;
```

This cookie is correctly set.

![image](https://github.com/lovasoa/SQLpage/assets/136404184/7edf05f8-90ca-4b98-a5d5-f5315570830d)

And then in the page that it redirects to, I do this:

```sql
-- clear flash cookie
select 'cookie' as component, 'admin.alert.title' as name, true as remove;

--- display the value
select 'alert' as component,
	sqlpage.cookie('admin.alert.title') as title,
	'green' as color
	where sqlpage.cookie('admin.alert.title') is not null;
```

The alert is correctly displayed, and I can see that the HTTP headers are trying to unset the cookie in the response (expiration date is in the past):

![image](https://github.com/lovasoa/SQLpage/assets/136404184/04671411-a91c-40bd-8e8d-060d97c31793)

Frustratingly, Firefox and Chrome are both hanging on to the cookie and thus I'm continuing to see the alert. I don't think this is a technical problem with SQLPage because it looks like the correct HTTP header is being sent to remove the cookie, but I just can't wrap my mind around why this isn't working. In desperation I tried this, which *does* work.

```
select 'cookie' as component, 'admin.alert.title' as name, unixepoch() as expires, '' as value;
```

My computer clock is correct, so I am baffled at why `true as remove` doesn't work...

**Comment by E8y2FqZE** [comment]
Could it be that the `Path` is provided in the HTTP header when setting the cookie (default of `/` because I didn't specify one) but NOT provided when `true as remove` is set? If so, maybe this could be a SQLPage bug.

@lovasoa If the missing Path is the problem, then this early return could be the cause: https://github.com/lovasoa/SQLpage/blob/cef4066e872dd76b76cda3fcc43ff1088fdb5ddd/src/render.rs#L122

**Reply by lovasoa** [reply]
Yes, this looks like a bug with the cookie's path. Could you make a pull request?

**Reply by lovasoa** [reply, marked as answer]
I made the fix myself in https://github.com/lovasoa/SQLpage/commit/6c68d1db2b04c8483770fbb8dccb67a9f0f1ee7f

Thanks for reporting the issue, @E8y2FqZE !

**Comment by lovasoa** [comment]
v0.23 was released, with the fix: https://github.com/lovasoa/SQLpage/releases/tag/v0.23.0
